File: src/main/resources/output/IFC4X3_java/Types/IfcClassificationReferenceSelect.java
// SELECT TYPE IfcClassificationReferenceSelect
public class IfcClassificationReferenceSelect {
    public Object value;

    public IfcClassificationReferenceSelect(Object value) {
        if (!(value instanceof IfcClassification || value instanceof IfcClassificationReference)) {
            throw new IllegalArgumentException("Value must be one of: IfcClassification, IfcClassificationReference");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}

File: src/main/resources/output/IFC4X3_js/Types/IfcClassificationReferenceSelect.js
// SELECT TYPE IfcClassificationReferenceSelect
// Options:
// - IfcClassification
// - IfcClassificationReference
class IfcClassificationReferenceSelect {
    constructor(value) {
        if (!(value instanceof IfcClassification || value instanceof IfcClassificationReference)) {
            throw new Error("Value must be one of: IfcClassification, IfcClassificationReference");
        }
        this.value = value;
    }

    getValueAs(TypeClass) {
        if (this.value instanceof TypeClass) {
            return this.value;
        } else {
            throw new Error("Stored value is not of the requested type");
        }
    }
}
